src/renderers/shared/fiber/__tests__/ReactIncrementalUpdates-test.js

```diff
@@ -0,0 +1,320 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+var React;
+var ReactNoop;
+
+describe('ReactIncrementalUpdates', () => {
+  beforeEach(() => {
+    jest.resetModuleRegistry();
+    React = require('React');
+    ReactNoop = require('ReactNoop');
+  });
+
+  function div(...children) {
+    children = children.map(c => typeof c === 'string' ? { text: c } : c);
+    return { type: 'div', children, prop: undefined };
+  }
+
+  function span(prop) {
+    return { type: 'span', children: [], prop };
+  }
+
+  it('applies updates in order of priority', () => {
+    let state;
+    class Foo extends React.Component {
+      state = {};
+      componentDidMount() {
+        ReactNoop.performAnimationWork(() => {
+          //  Has Animation priority
+          this.setState({ b: 'b' });
+          this.setState({ c: 'c' });
+        });
+        // Has Task priority
+        this.setState({ a: 'a' });
+      }
+      render() {
+        state = this.state;
+        return <div />;
+      }
+    }
+
+    ReactNoop.render(<Foo />);
+    ReactNoop.flush();
+    expect(Object.keys(state)).toEqual(['a', 'b', 'c']);
```

Comment:
We shouldn't rely on object iteration order. This has failed us in the past, and in fact, we have talked about enforcing that all keys are always defined to enforce that the state object has a consistent hidden class.

Can you rewrite this to test that there are actually two renders and that the first one only have part of the state and the second one has all three?

This same goes for all the other tests in this file that use the same hack.